synapse/tools/dmon.py

```diff
@@ -143,9 +144,14 @@ def main(argv, outp=None):
     p = getArgParser()
     opts = p.parse_args(argv)
 
-    if opts.log_level:
-        logging.basicConfig(level=opts.log_level.upper())
-        logger.info('log level set to ' + opts.log_level)
+    log_level = os.getenv('SYN_DMON_LOG_LEVEL', opts.log_level)
+    if log_level:
+        log_level = log_level.upper()
+        if log_level not in LOG_LEVEL_CHOICES:
+            raise Exception('Invalid log level provided: {}'.format(log_level))
```

Comment:
Do we have a config error type?

Reply:
we could unify exceptions being thrown in dmon.

Reply:
We have BadConfValu() but since it's not an actual Config value, i might just stick with something generic ( unless we wanna make it an official part of the config? )

Reply:
since the dmon tool can take in multiple configs at once, we have a multi-master problem if we plumb the logging from the config files.

Reply:
this could easily be a ValueError